<?xml version="1.0"?>

<launch>
  <param name="/use_sim_time" value="true" />

  <!-- Please set up the tf of base_link_to_laser according to your datasets -->
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_laser"
    args="0.0 0.0 0.0 0.0 0.0 0.0 /base_link /laser 40" />

  <!-- Please set up the scan topic and odom frame according to your datasets -->
  <node pkg="slam_karto_g2o" type="slam_karto_g2o" name="slam_karto_g2o" output="screen">
    <param name="scan_topic" value="scan"/>
    <param name="odom_frame" value="odom"/>
    <param name="map_update_interval" value="5.0"/>
    <param name="resolution" value="0.025"/>
    <param name="range_threshold" value="12.0"/>
    <rosparam command="load" file="$(find slam_karto_g2o)/config/mapper_params.yaml" />
  </node>

  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find slam_karto_g2o)/launch/demo.rviz"/>
</launch>
